source/scenes/shop.js

```diff
@@ -0,0 +1,268 @@
+    //         updateShop();
+    //     }
+    // }
+
+    loadShop(type) {
+        // Loads the correct background image (changes which button is selected)
+        this.add.image(0, 0, 'shop_'+type).setOrigin(0,0).setDepth(0);
+
+        // Displays cash stack 
+        this.displayGold(this.characterData.gold);
+
+        // Loads items into shopItems, dynamically loads images for the items, and displays items on screen
+        this.loadItems(type, this.characterData.skills);
+    }
+
+    // Outputs the gold text in RS format: 1m = 1000k, 10m = green text, 1b = 1000m
+    displayGold(gold) {
+        // Pick gold image based on # of coins 
+        let stack = '';
+        switch(true) {
+            case (gold < 25):
+                stack = '5'; break;
+            case (gold < 100):
+                stack = '25'; break;
+            case (gold < 250):
+                stack = '100'; break;
+            case (gold < 1000):
+                stack = '250'; break;
+            case (gold < 10000):
+                stack = '1k'; break;
+            default:
+                stack = '10k'; break;
+        }
+        let goldImage = this.add.image(48, 346, stack).setInteractive();
+        goldImage.scale = 1.3;
+
+        // Pick text color and style based on # of coins
+        let color = 'white';
+        let goldText = this.characterData.gold;
+        if(gold > 99999 && gold < 10000000) {
+            goldText = (gold / 1000) + 'k';
+        }
+        else if(gold > 10000000) {
+            goldText = (gold / 1000000) + 'M';
+            fill = '#06c663';
+        }
+        this.add.text(33, 320, goldText, {fontFamily: '"runescape_uf"', fill: color, stroke: '#000000', strokeThickness: 2});
+    }
+
+    // Load items based on the character's levels
+    loadItems(itemType, characterSkills) {
+        console.log('Loading ' + itemType);
+        this.loadingText.visible = true;
+
+        // Reset the shop when loading a new type of item
+        this.shopItems = [], this.shopIcons = [], this.displayIndex = 0;
+
+        let imageX = {}, imageY = {};
+        switch(itemType) {
+            // Weapons organized with attack on row 1, ranged on row 2, magic on row 3
+            case 'WEAPONS':
+                imageX = {'attack': 100, 'ranged': 100, 'magic': 100};
+                imageY = {'attack': 100, 'ranged': 200, 'magic': 300};
+                this.loadCount++;
+                break;
+            // Tools organized with all tools on one row
+            case 'TOOLS':
+                imageX = {'woodcutting': 100, 'mining': 200};
+                imageY = {'woodcutting': 100, 'mining': 100};
+                break;
+            // Consumables organized by food on row 1, potions on row 2
+            // @TODO: This organization scheme will need to be changed in the future
+            case 'CONSUMABLES':
+                imageX = {'health': 100, 'herblore': 100};
+                imageY = {'health': 100, 'herblore': 200};
```

Comment:
FYI you don't need quotes around 'attack', 'ranged', etc., imageX['attack'] will work regardless